app/email.go

```diff
@@ -170,6 +170,10 @@ func (a *App) SendSignInChangeEmail(email, method, locale, siteURL string) *mode
 }
 
 func (a *App) SendWelcomeEmail(userId string, email string, verified bool, locale, siteURL string) *model.AppError {
+	if !*a.Config().EmailSettings.SendEmailNotifications {
```

Comment:
I'm not sure, but I think this email should be sent if this is for verification but not for welcome. If you require email verification, but doesn't have email notifications, is normal to not receive the welcome message, but the email verification is needed for the application functionality, not for notifying the user. Any opinion on this @wiersgallak?

Reply:
@jespino Already discussed this with @thefactremains . He mentioned that it should still be sent even if Email Notifications is off. I made a mistake. Here is the possibilities chart
```
Email Verification Setting | Email Notification Setting
0 0 - No Email Sent
1 0 - Email with Verification link sent
0 1 - Welcome Email (no verification link) sent
1 1 - Email with Verification link sent
```
I will fix this shortly and push commit.